WordPress/Classes/ViewRelated/Pages/BasePageListCell.h

```diff
@@ -1,19 +1,33 @@
 #import <UIKit/UIKit.h>
 #import "WPPostContentViewProvider.h"
 
+@class AbstractPost;
+@class BasePageListCell;
 
-@protocol BasePageListCellDelegate <NSObject>
-@optional
-- (void)cell:(nonnull UITableViewCell *)cell receivedMenuActionFromButton:(nonnull UIButton *)button forProvider:(nonnull id<WPPostContentViewProvider>)contentProvider;
-- (void)cell:(nonnull UITableViewCell *)cell receivedRestoreActionForProvider:(nonnull id<WPPostContentViewProvider>)contentProvider;
-@end
-
+/// A block that represents an action triggered by tapping on a button in a cell.
+///
+/// @param      cell        The cell that contains the button that was tapped.
+/// @param      button      The button that was tapped.
+/// @param      post        The post represented by the cell that was tapped.
+///
+typedef void(^BasePageListCellActionBlock)(BasePageListCell* _Nonnull cell,
```

Comment:
You can't put the `_Nonnull` attribute before the `*`.  I first tried using the prefix `nonnull` attribute but that didn't seem to work for Xcode.

Let me know if you know of a way to make this look better / or prefer it differently.  I'd suggest trying it out in Xcode too though because it seems like some of the rules used for method parameters don't work the same in blocks.